### PR TITLE
refactor(migrations): initial migration logic for converting to signal queries

### PR DESCRIPTION
**Pull request by devversion**
Adds initial migration logic to convert decorator query declarations to signal queries.

We will re-use more part of the signal input migration in follow-ups, to properly migrate,
and e.g. even handle control flow

### CODE REVIEW

File: packages/core/schematics/migrations/signal-queries-migration/convert_query_property.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    requestedFile: sf,
+    exportModuleSpecifier: '@angular/core',
+    exportSymbolName: metadata.kind,
+  });
+
+  // The default value for descendants is `true`, except for `ContentChildren`.
+  const defaultDescendants = metadata.kind !== 'contentChildren';
+  const optionProperties: ts.PropertyAssignment[] = [];
+  const args: ts.Expression[] = [
+    metadata.args[0], // Locator.
```
Comment (crisbeto) [inline packages/core/schematics/migrations/signal-queries-migration/convert_query_property.ts:57]: Isn't it possible for this to be undefined?

**Review by crisbeto — APPROVED** [review]

File: packages/core/schematics/migrations/signal-queries-migration/convert_query_property.ts (hunk 1 above)
Comment (devversion) [reply packages/core/schematics/migrations/signal-queries-migration/convert_query_property.ts:57]: Probably not. I will add some safety fallback to `undefined` then.

Comment (devversion) [reply packages/core/schematics/migrations/signal-queries-migration/convert_query_property.ts:57]: Seems like it's guaranteed to exist given this is metadata parsed by the Angular compiler.

File: packages/core/schematics/migrations/signal-queries-migration/identify_queries.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+} from '../../../../compiler-cli/src/ngtsc/annotations';
+import {ReflectionHost} from '../../../../compiler-cli/src/ngtsc/reflection';
+import {UniqueID} from '../../utils/tsurge';
+import path from 'path';
+import {extractDecoratorQueryMetadata} from '../../../../compiler-cli/src/ngtsc/annotations/directive';
+import {PartialEvaluator} from '../../../../compiler-cli/private/migrations';
+import {R3QueryMetadata} from '../../../../compiler';
+
+/** Branded type to uniquely identify class properties in a project. */
+export type ClassPropertyID = UniqueID<'ClassPropertyID; Potentially an output'>;
```
Comment (pkozlowski-opensource) [inline packages/core/schematics/migrations/signal-queries-migration/identify_queries.ts:22]: Potentially an output => Potentially a query?

Comment (devversion) [reply packages/core/schematics/migrations/signal-queries-migration/identify_queries.ts]: good catch. Fixed

**Comment by alxhub** [comment]
This PR was merged into the repository by commit f454ad3bcff801895d8fa1e4316bdb5da8b7368c.

The changes were merged into the following branches: main, 18.2.x

**Pull request closed without merging**